### PR TITLE
goplus 1.0.32 (new formula)

### PR DESCRIPTION
**Pull request by cpunion**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/goplus.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  depends_on "go"
+
+  def install
+    # Patch version to match the version of gop
+    unless build.head?
+      version_content = File.read("env/version.go")
+      version_content.sub!(/^\tbuildVersion string$/, "\t	buildVersion string = \"v#{version}\"")
+      File.open("env/version.go", "w") { |file| file.puts version_content }
+    end
```
Comment (carlocab) [inline Formula/goplus.rb:18]: Can't we use `inreplace` here?

File: Formula/goplus.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    ENV.prepend "GOPROOT_FINAL", libexec
```
Comment (carlocab) [inline Formula/goplus.rb:20]: Why not
```suggestion
    ENV["GOPROOT_FINAL"] = libexec
```
?

File: Formula/goplus.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+class Goplus < Formula
+  desc "Programming language for engineering, STEM education, and data science"
+  homepage "https://goplus.org"
+  url "https://github.com/goplus/gop/archive/refs/tags/v1.0.32.tar.gz"
+  sha256 "c008d6fcc8c8f3a08e69774a6e54dce82fae78414b6e5c45b5b619dd0572a139"
+  license "Apache-2.0"
+
```
Comment (carlocab) [inline Formula/goplus.rb:7]:
```suggestion
```

File: Formula/goplus.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    ENV.prepend "GOPROOT_FINAL", libexec
+    system "go", "run", "cmd/install.go", "--install"
+
+    Dir["*"].each do |f|
+      next if f.start_with?(".")
+
+      libexec.install f
+    end
+
+    bin.install_symlink Dir[libexec/"bin/*"]
```
Comment (carlocab) [inline Formula/goplus.rb:29]:
```suggestion
    libexec.install Dir["*"] - Dir[".*"]
    bin.install_symlink Dir[libexec/"bin/*"]
```

File: Formula/goplus.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      assert_equal "Hello World\n", shell_output("#{bin/"gop"} run hello.gop")
+
+      (testpath/"go.mod").write <<~EOS
+        module hello
+      EOS
+
+      system "go", "get", "github.com/goplus/gop/builtin"
+      system bin/"gop", "build", "-o", "hello"
+      assert_equal "Hello World\n", shell_output("./hello")
+    end
```
Comment (carlocab) [inline Formula/goplus.rb:54]:
```suggestion
    # Run gop fmt, run, build
    ENV.prepend "GO111MODULE", "on"

    assert_equal "v#{version}", shell_output("#{bin}/gop env GOPVERSION").chomp unless head?
    system bin/"gop", "fmt", "hello.gop"
    assert_equal "Hello World\n", shell_output("#{bin}/gop run hello.gop")
    
    (testpath/"go.mod").write <<~EOS
      module hello
    EOS

    system "go", "get", "github.com/goplus/gop/builtin"
    system bin/"gop", "build", "-o", "hello"
    assert_equal "Hello World\n", shell_output("./hello")
```

File: Formula/goplus.rb (hunk 1 above)
Comment (cpunion) [reply Formula/goplus.rb:18]: Yeh, I got it.

**Comment by cpunion** [comment]
@carlocab Updated, thanks for your review.

File: Formula/goplus.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  depends_on "go"
+
+  def install
+    # Patch version to match the version of gop, currently it get version from git tag
+    inreplace "env/version.go", /^\tbuildVersion string$/, "\tbuildVersion string = \"v#{version}\"" unless build.head?
+
+    build_script = "cmd/install.go"
+    # cmd/install.go renamed to cmd/make.go in the main branch
+    build_script = "cmd/make.go" if build.head?
```
Comment (carlocab) [inline Formula/goplus.rb:17]:
```suggestion
    inreplace "env/version.go", /^\tbuildVersion string$/, "\tbuildVersion string = \"v#{version}\"" unless build.head?
    build_script = build.head? ? "cmd/make.go" : "cmd/install.go"
```

**Review by carlocab — APPROVED** [review]
Thanks for your contribution!

**Pull request closed without merging**